src/main/java/org/reactome/orthoinference/InstanceUtilities.java

```diff
@@ -204,6 +205,59 @@ public static String getCacheKey(GKSchemaClass instanceClass, GKInstance inferre
 		}
 		return key;
 	}
+
+	/**
+	 * This method returns true if the only parent Pathway of the incoming Event instane is Disease.
+	 * Instances with only Disease as a parent will not be inferred. Those that are a member of Disease AND
+	 * another Pathway will be inferred, but the Disease Pathway (and sub-Pathway) inference will be skipped.
+	 * @param eventInst -- GKInstance that will be checked for parent Pathways.
+	 * @return boolean -- true if only parent is Disease Pathway, false if not.
+	 * @throws Exception -- Thrown by MySQLAdaptor.
+	 */
+	public static boolean onlyInDiseasePathway(GKInstance eventInst) throws Exception {
+		Set<Long> topLevelPathwayDbIds = getTopLevelParents(eventInst);
+		return topLevelPathwayDbIds.size() == 1 && topLevelPathwayDbIds.contains(DISEASE_PATHWAY_DBID);
+	}
+
+	/**
+	 * Helper method for 'onlyInDiseasePathway'. Finds DbIds of all TopLevelPathways of instance.
+	 * @param eventInst -- GKInstance for which all TopLevelPathway DbIds will be found.
+	 * @return -- Set<GKInstance> containing a list of all TopLevelPathway DbIds for incoming Event instance.
+	 * @throws Exception -- Thrown by MySQLAdaptor.
+	 */
+	private static Set<Long> getTopLevelParents(GKInstance eventInst) throws Exception {
+		Set<Long> topLevelPathwayDbIds = new HashSet<>();
+		if (eventInst.getReferers(hasEvent) != null) {
+			for (GKInstance pathwayInst : (Collection<GKInstance>) eventInst.getReferers(hasEvent)) {
+				topLevelPathwayDbIds.addAll(getPathwayReferrals(pathwayInst));
+			}
+		}
```

Comment:
For this check, I think we have a safeList utility method in the release-common-lib package you could use to wrap `eventInst.getReferers(hasEvent)` in the for loop.  That should allow you to remove the null check as well as the cast to `Collection<GKInstance>` (I think :))

Reply:
As far as I can tell, I will still need to cast the Collection

Reply:
Also, I was wondering how is this method different from the logic and purpose of `getPathwayReferrals` below?